src/main/java/hu/oe/nik/szfmv/automatedcar/model/XmlParser.java

```diff
@@ -0,0 +1,185 @@
+package hu.oe.nik.szfmv.automatedcar.model;
+
+import hu.oe.nik.szfmv.automatedcar.model.objects.CrossWalk;
+import hu.oe.nik.szfmv.automatedcar.model.objects.ParkingPlace;
+import hu.oe.nik.szfmv.automatedcar.model.objects.Tree;
+import hu.oe.nik.szfmv.automatedcar.model.objects.Road;
+import hu.oe.nik.szfmv.automatedcar.model.objects.RoadSign;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import javax.xml.crypto.dsig.XMLSignatureException;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class XmlParser {
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public static List<WorldObject> build(String xmlLocation)
+        throws ParserConfigurationException, IOException, SAXException {
+        Road.loadReferencePoints();
+
+        List<WorldObject> objectListToReturn = new ArrayList<WorldObject>();
+        File inputFile = new File(xmlLocation);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder dBuilder = factory.newDocumentBuilder();
+        Document doc = dBuilder.parse(inputFile);
+
+        NodeList xmlNodes = doc.getElementsByTagName("Object");
+
+        for (int iterator = 0; iterator < xmlNodes.getLength(); iterator++) {
+            Node nodeToAdd = xmlNodes.item(iterator);
+            try {
+                objectListToReturn.add(readValueFromXml((Element) nodeToAdd));
+            } catch (Exception e) {
+
+                LOGGER.error("XML object parse Error in object #" + iterator + ": " + e.getMessage());
+            }
+        }
+        return objectListToReturn;
+    }
+
+    /**
+     * @param objectElement A Xml Object that can be converted to WorldObject
+     * @return WorldObject created from Xml Object
+     * @throws XMLSignatureException thrown if tag missing from XmlObject
+     * @throws IOException           thrown if WorldObject representative picture not found
+     */
+    private static WorldObject readValueFromXml(Element objectElement)
```

Comment:
Ez azért egy elég hosszú metódus.

Lehet, hogy én értek valamit félre, de ez a logika teljesen nyakatekert. Be lehet járni úgy is XML-t, hogy csak adott node-okon (pl. „Object”) iterálsz (lásd még XPath). Szerintem még Javában is lehet olyat, hogy ha már megvan egy elemed, akkor annak egy adott nevű gyermekét olvasod ki (ez térhet vissza null-lal)

Az viszont egyértelmű, hogy ez a metódus 2 dolgot csinál. Kiolvassa az elem tulajdonságait (readValueFromXml, erre jó a név), majd példányosít egy WorldObjectet.